data/pigui/modules/new-game-window/class.lua

```diff
@@ -0,0 +1,231 @@
+-- Copyright © 2008-2023 Pioneer Developers. See AUTHORS.txt for details
+-- Licensed under the terms of the GPL v3. See licenses/GPL-3.txt
+
+local Lang = require 'Lang'
+local ui = require 'pigui'
+local lui = Lang.GetResource("ui-core")
+local msgbox = require 'pigui.libs.message-box'
+local Character = require 'Character'
+local Commodities = require 'Commodities'
+local Equipment = require 'Equipment'
+local FlightLog = require 'FlightLog'
+local ModalWindow = require 'pigui.libs.modal-win'
+local ModelSkin = require 'SceneGraph.ModelSkin'
+local ShipDef = require "ShipDef"
+
+local Defs = require 'pigui.modules.new-game-window.defs'
+local Layout = require 'pigui.modules.new-game-window.layout'
+local Crew = require 'pigui.modules.new-game-window.crew'
+local startVariants = require 'pigui.modules.new-game-window.start-variants'
+local Game = require 'Game'
+
+local profileCombo = { items = {}, selected = 0 }
+
+-- pass avalable space rect
+local function updateLayout(contentRegion)
+	Layout.updateLayout(contentRegion)
+
+	for _, tab in ipairs(Layout.Tabs) do
+		tab:updateLayout()
+	end
+end
+
+local function setStartVariant(variant)
+	for _, param in ipairs(Layout.UpdateOrder) do
+		param:fromStartVariant(variant)
+	end
+end
+
+local function startGame(gameParams)
```

Comment:
Ideally this function should provide a way for modules which provide additional start variants etc. to perform initial setup on the player's ship / universe state, perhaps by calling an event handler and passing the `gameParams` value.